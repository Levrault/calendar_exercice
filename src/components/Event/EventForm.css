.container {
  background-color: $palette-text-or-icon;
  border-radius: $space-s;
  bottom: resolve($space-l + $space-xs);
  box-shadow: 0 10px 20px rgba(0,0,0,.19), 0 6px 6px rgba(0,0,0,.23);
  left: -75px;
  padding: $space-m;
  position: absolute;
  z-index: $foreground;
}

.eventForm {
  min-width: 150px;
}

.header {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.buttons {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.colorPalette {
  @mixin horizontalList;
  flex-wrap: wrap;
  margin-bottom: $space-s;
  max-height: 0;
  opacity: 0;
  overflow-y: hidden;
  transition-duration: 1.5s;
  transition-property: all;
  transition-timing-function: cubic-bezier(0, 1, .5, 1);

  &.active {
    height: resolve($space-l * 4);
    max-height: resolve($space-l * 4);
    opacity: 1;
  }
}

.color {
  display: flex;
  flex-grow: 1;
  justify-content: center;
  margin-top: $space-s;
  width: 25%;
}
