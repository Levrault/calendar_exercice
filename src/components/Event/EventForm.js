import React, { Component } from 'react';
import PropTypes from 'prop-types';
import EventButton from '../Button/EventButton';
import ColorInputField from '../Field/ColorInputField';
import ColorPalette from '../Palette/ColorPalette';
import colors from '../Palette/colors-constant';
import Message from '../Message/Message';
import styles from './EventForm.css';

class EventForm extends Component {
  /**
  * @contructor
  * @param {object} props
  */
  constructor (props) {
    super(props);
    this.state = {
      displayColorField: false,
      name: '',
      color: colors[colors.length - 1].code
    };
  }

  componentWillUnmount () {
    this.props.handleReset();
  }

  /**
   * Show/hide color palette
   */
  handleColorPaletteClick = () => {
    this.setState({
      displayColorField: !this.state.displayColorField
    });
  }

  /**
   * @param {object} event
   */
  handleColorChange = (event) => {
    const value = event.target.value;
    this.props.handleColorChange(value);
    this.setState({
      color: value
    });
  }

  /**
   * @param {object} event
   */
  handleNameChange = (event) => {
    console.log('event.target', event.target); //@TODO : to remove
    this.setState({
      name: event.target.value
    });
  }

  /**
   * Close form
   */
  handleCloseClick = () => {
    this.props.onCancel();
  }

  /**
   * Submit form
   * @param {object} event
   */
  handleSubmit = (event) => {
    event.preventDefault();
    const { name, color } = this.state;
    const { monthId, day } = this.props;
    this.props.handleSubmit({
      name,
      color,
      monthId,
      day
    });
  }

  /**
  * Render
  */
  render () {
    const { displayColorField, color, name } = this.state;
    const { success, errors } = this.props;

    return (
      <div className={styles.container}>
        <form className={styles.eventForm} onSubmit={this.handleSubmit}>

          <div className={styles.header}>
            <ColorInputField
              name="event"
              type="text"
              label="Rappel"
              value={name}
              onChange={this.handleNameChange}
            />
          </div>

          <ColorPalette active={displayColorField} onClick={this.handleColorChange} value={color} />

          {success && <Message type="success">Event added</Message>}

          {errors &&
              Object.keys(errors).map(index => {
                const error = errors[index];
                return <Message key={error} type="error">{index} {error}</Message>;
              })
          }

          <div className={styles.buttons}>

            <EventButton type="button" onClick={this.handleCloseClick}>
              <i className="material-icons">clear</i>
            </EventButton>

            <EventButton type="submit">
              <i className="material-icons">add</i>
            </EventButton>

            <EventButton type="button" onClick={this.handleColorPaletteClick} active={displayColorField}>
              <i className="material-icons">color_lens</i>
            </EventButton>
          </div>

        </form>
      </div>
    );
  }
}

EventForm.propTypes = {
  onCancel: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleColorChange: PropTypes.func.isRequired,
  handleReset: PropTypes.func.isRequired,
  monthId: PropTypes.string.isRequired,
  day: PropTypes.number.isRequired,
  errors: PropTypes.object,
  success: PropTypes.bool
};

EventForm.defaultProps = {
  currentColor: colors[colors.length - 1].code,
  success: false
};

export default EventForm;
