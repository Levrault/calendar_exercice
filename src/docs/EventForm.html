<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: EventForm</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: EventForm</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>EventForm<span class="signature">(onCancel, onSubmit, onColorChange, onReset, monthId, day, errors<span class="signature-attributes">opt</span>)</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="EventForm"><span class="type-signature"></span>new EventForm<span class="signature">(onCancel, onSubmit, onColorChange, onReset, monthId, day, errors<span class="signature-attributes">opt</span>)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Event form
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        
        <th>Attributes</th>
        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>onCancel</code></td>
            

            <td class="type">
            
                
<span class="param-type">function</span>


            
            </td>

            
                <td class="attributes">
                

                

                
                </td>
            

            

            <td class="description last"></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>onSubmit</code></td>
            

            <td class="type">
            
                
<span class="param-type">function</span>


            
            </td>

            
                <td class="attributes">
                

                

                
                </td>
            

            

            <td class="description last"></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>onColorChange</code></td>
            

            <td class="type">
            
                
<span class="param-type">function</span>


            
            </td>

            
                <td class="attributes">
                

                

                
                </td>
            

            

            <td class="description last"></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>onReset</code></td>
            

            <td class="type">
            
                
<span class="param-type">function</span>


            
            </td>

            
                <td class="attributes">
                

                

                
                </td>
            

            

            <td class="description last"></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>monthId</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            
                <td class="attributes">
                

                

                
                </td>
            

            

            <td class="description last"></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>day</code></td>
            

            <td class="type">
            
                
<span class="param-type">number</span>


            
            </td>

            
                <td class="attributes">
                

                

                
                </td>
            

            

            <td class="description last"></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>errors</code></td>
            

            <td class="type">
            
                
<span class="param-type">object</span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                </td>
            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Event_EventForm.js.html">components/Event/EventForm.js</a>, <a href="components_Event_EventForm.js.html#line10">line 10</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    

    

    

    

    

    

    
        <h3 class="subsection-title">Members</h3>

        
            
<h4 class="name" id="handleCloseClick"><span class="type-signature"></span>handleCloseClick<span class="type-signature"></span></h4>




<div class="description">
    Close event form
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Event_EventForm.js.html">components/Event/EventForm.js</a>, <a href="components_Event_EventForm.js.html#line79">line 79</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="handleColorChange"><span class="type-signature"></span>handleColorChange<span class="type-signature"></span></h4>




<div class="description">
    Update color's value
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Event_EventForm.js.html">components/Event/EventForm.js</a>, <a href="components_Event_EventForm.js.html#line58">line 58</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="handleColorPaletteClick"><span class="type-signature"></span>handleColorPaletteClick<span class="type-signature"></span></h4>




<div class="description">
    Show/hide color palette
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Event_EventForm.js.html">components/Event/EventForm.js</a>, <a href="components_Event_EventForm.js.html#line46">line 46</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="handleNameChange"><span class="type-signature"></span>handleNameChange<span class="type-signature"></span></h4>




<div class="description">
    Update name's value
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Event_EventForm.js.html">components/Event/EventForm.js</a>, <a href="components_Event_EventForm.js.html#line70">line 70</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="onSubmit"><span class="type-signature"></span>onSubmit<span class="type-signature"></span></h4>




<div class="description">
    Submit form
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Event_EventForm.js.html">components/Event/EventForm.js</a>, <a href="components_Event_EventForm.js.html#line87">line 87</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="componentWillUnmount"><span class="type-signature"></span>componentWillUnmount<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Reset form when mounted
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Event_EventForm.js.html">components/Event/EventForm.js</a>, <a href="components_Event_EventForm.js.html#line39">line 39</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
            

    

    
    <h4 class="name" id="render"><span class="type-signature"></span>render<span class="signature">()</span><span class="type-signature"> &rarr; {node}</span></h4>
    

    



<div class="description">
    Render
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Event_EventForm.js.html">components/Event/EventForm.js</a>, <a href="components_Event_EventForm.js.html#line104">line 104</a>
    </li></ul></dd>
    

    

    

    
</dl>













<h5>Returns:</h5>

        


<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">node</span>


    </dd>
</dl>

    





        
    

    

    
</article>

</section>







<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>EventForm<span class="signature">(props)</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="EventForm"><span class="type-signature"></span>new EventForm<span class="signature">(props)</span><span class="type-signature"></span></h4>
    

    











    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>props</code></td>
            

            <td class="type">
            
                
<span class="param-type">object</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Event_EventForm.js.html">components/Event/EventForm.js</a>, <a href="components_Event_EventForm.js.html#line26">line 26</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    

    

    

    

    

    

    
        <h3 class="subsection-title">Members</h3>

        
            
<h4 class="name" id="handleCloseClick"><span class="type-signature"></span>handleCloseClick<span class="type-signature"></span></h4>




<div class="description">
    Close event form
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Event_EventForm.js.html">components/Event/EventForm.js</a>, <a href="components_Event_EventForm.js.html#line79">line 79</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="handleColorChange"><span class="type-signature"></span>handleColorChange<span class="type-signature"></span></h4>




<div class="description">
    Update color's value
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Event_EventForm.js.html">components/Event/EventForm.js</a>, <a href="components_Event_EventForm.js.html#line58">line 58</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="handleColorPaletteClick"><span class="type-signature"></span>handleColorPaletteClick<span class="type-signature"></span></h4>




<div class="description">
    Show/hide color palette
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Event_EventForm.js.html">components/Event/EventForm.js</a>, <a href="components_Event_EventForm.js.html#line46">line 46</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="handleNameChange"><span class="type-signature"></span>handleNameChange<span class="type-signature"></span></h4>




<div class="description">
    Update name's value
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Event_EventForm.js.html">components/Event/EventForm.js</a>, <a href="components_Event_EventForm.js.html#line70">line 70</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="onSubmit"><span class="type-signature"></span>onSubmit<span class="type-signature"></span></h4>




<div class="description">
    Submit form
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Event_EventForm.js.html">components/Event/EventForm.js</a>, <a href="components_Event_EventForm.js.html#line87">line 87</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="componentWillUnmount"><span class="type-signature"></span>componentWillUnmount<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Reset form when mounted
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Event_EventForm.js.html">components/Event/EventForm.js</a>, <a href="components_Event_EventForm.js.html#line39">line 39</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
            

    

    
    <h4 class="name" id="render"><span class="type-signature"></span>render<span class="signature">()</span><span class="type-signature"> &rarr; {node}</span></h4>
    

    



<div class="description">
    Render
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Event_EventForm.js.html">components/Event/EventForm.js</a>, <a href="components_Event_EventForm.js.html#line104">line 104</a>
    </li></ul></dd>
    

    

    

    
</dl>













<h5>Returns:</h5>

        


<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">node</span>


    </dd>
</dl>

    





        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="App.html">App</a></li><li><a href="Calendar.html">Calendar</a></li><li><a href="ColorPalette.html">ColorPalette</a></li><li><a href="Day.html">Day</a></li><li><a href="EventForm.html">EventForm</a></li><li><a href="EventShow.html">EventShow</a></li><li><a href="Month.html">Month</a></li><li><a href="YearForm.html">YearForm</a></li></ul><h3>Global</h3><ul><li><a href="global.html#About">About</a></li><li><a href="global.html#CalendarButton">CalendarButton</a></li><li><a href="global.html#checkValidServiceWorker">checkValidServiceWorker</a></li><li><a href="global.html#ColorButton">ColorButton</a></li><li><a href="global.html#colorChange">colorChange</a></li><li><a href="global.html#ColorInputField">ColorInputField</a></li><li><a href="global.html#dayReducer">dayReducer</a></li><li><a href="global.html#daySelector">daySelector</a></li><li><a href="global.html#deleteEvent">deleteEvent</a></li><li><a href="global.html#Error">Error</a></li><li><a href="global.html#EventButton">EventButton</a></li><li><a href="global.html#eventCreated">eventCreated</a></li><li><a href="global.html#eventDeleted">eventDeleted</a></li><li><a href="global.html#eventFormDeleteBegin">eventFormDeleteBegin</a></li><li><a href="global.html#eventFormDeleteError">eventFormDeleteError</a></li><li><a href="global.html#eventFormDeleteSuccess">eventFormDeleteSuccess</a></li><li><a href="global.html#eventFormReducer">eventFormReducer</a></li><li><a href="global.html#eventFormReset">eventFormReset</a></li><li><a href="global.html#eventFormSelector">eventFormSelector</a></li><li><a href="global.html#eventFormSubmitBegin">eventFormSubmitBegin</a></li><li><a href="global.html#eventFormSubmitError">eventFormSubmitError</a></li><li><a href="global.html#eventFormSubmitSuccess">eventFormSubmitSuccess</a></li><li><a href="global.html#eventsInit">eventsInit</a></li><li><a href="global.html#eventsReducer">eventsReducer</a></li><li><a href="global.html#eventsSelector">eventsSelector</a></li><li><a href="global.html#fetch">fetch</a></li><li><a href="global.html#fetchAll">fetchAll</a></li><li><a href="global.html#Link">Link</a></li><li><a href="global.html#Loader">Loader</a></li><li><a href="global.html#Main">Main</a></li><li><a href="global.html#mapStateToProps">mapStateToProps</a></li><li><a href="global.html#Message">Message</a></li><li><a href="global.html#Navbar">Navbar</a></li><li><a href="global.html#NavbarRouter">NavbarRouter</a></li><li><a href="global.html#onSelected">onSelected</a></li><li><a href="global.html#PageNotFound">PageNotFound</a></li><li><a href="global.html#post">post</a></li><li><a href="global.html#registerValidSW">registerValidSW</a></li><li><a href="global.html#render">render</a></li><li><a href="global.html#Title">Title</a></li><li><a href="global.html#unregister">unregister</a></li><li><a href="global.html#WeekHeader">WeekHeader</a></li><li><a href="global.html#yearFetchAllBegin">yearFetchAllBegin</a></li><li><a href="global.html#yearFetchAllError">yearFetchAllError</a></li><li><a href="global.html#yearFetchAllSuccess">yearFetchAllSuccess</a></li><li><a href="global.html#yearFetchBegin">yearFetchBegin</a></li><li><a href="global.html#yearFetchError">yearFetchError</a></li><li><a href="global.html#yearFetchSuccess">yearFetchSuccess</a></li><li><a href="global.html#yearFormReducer">yearFormReducer</a></li><li><a href="global.html#yearSelector">yearSelector</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Fri Sep 21 2018 15:33:44 GMT+0200 (CEST)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>