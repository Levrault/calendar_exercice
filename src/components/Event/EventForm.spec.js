import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from '../../tools/test/enzyme';
import colors from '../Palette/colors-constant';
import EventForm from './EventForm';

const mock = {
  onCancel: jest.fn(),
  handleSubmit: jest.fn(),
  handleColorChange: jest.fn(),
  handleReset: jest.fn(),
  monthId: 'a4e8a03dbe54ca7d',
  day: 1,
  errors: null
};

describe('EventForm component', () => {
  it('should render without problem', () => {
    const component = renderer.create(
      <EventForm {...mock} errors={{ name: 'is required' }} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with errors', () => {
    const component = renderer.create(
      <EventForm {...mock} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call reset on unmount', () => {
    const component = shallow(
      <EventForm {...mock} />,
    );

    component.instance().componentWillUnmount();
    expect(component.instance().props.handleReset).toHaveBeenCalled();
  });

  it('should update displayColorField state when handleColorPaletteClick is called', () => {
    const component = shallow(
      <EventForm {...mock} />,
    );

    expect(component.instance().state.displayColorField).toBeFalsy();
    component.instance().handleColorPaletteClick();
    expect(component.instance().state.displayColorField).toBeTruthy();
  });

  it('should update color when handleColorChange is called', () => {
    const component = shallow(
      <EventForm {...mock} />,
    );

    expect(component.instance().state.color).toEqual(colors[colors.length - 1].code);
    component.instance().handleColorChange({ target: { value: colors[0].code } });
    expect(component.instance().state.color).toEqual(colors[0].code);
  });

  it('should update name when handleNameChange is called', () => {
    const component = shallow(
      <EventForm {...mock} />,
    );

    const value = 'name';
    expect(component.instance().state.name).toEqual('');
    component.instance().handleNameChange({ target: { value } });
    expect(component.instance().state.name).toEqual(value);
  });

  it('should call onCancel when handleCloseClick is called', () => {
    const component = shallow(
      <EventForm {...mock} />,
    );

    component.instance().handleCloseClick();
    expect(component.instance().props.onCancel).toHaveBeenCalled();
  });

  it('should call handleSubmit when handleSubmit is called', () => {
    const component = shallow(
      <EventForm {...mock} />,
    );
    component.instance().handleNameChange({ target: { value: 'name' } });
    component.instance().handleSubmit({ preventDefault: jest.fn() });
    expect(component.instance().props.handleSubmit).toHaveBeenCalled();
  });
});
