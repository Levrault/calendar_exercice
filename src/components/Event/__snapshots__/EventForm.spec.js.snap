// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventForm component should render with errors 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div>
      <div>
        <input
          name="event"
          onChange={[Function]}
          required={true}
          style={
            Object {
              "borderBottom": "2px solid undefined",
            }
          }
          type="text"
          value=""
        />
        <label
          htmlFor="nodejs"
        >
          Rappel
        </label>
      </div>
    </div>
    <ul
      className=""
    >
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#FF5252",
            }
          }
          type="button"
          value="#FF5252"
        />
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#FF4081",
            }
          }
          type="button"
          value="#FF4081"
        />
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E040FB",
            }
          }
          type="button"
          value="#E040FB"
        />
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#FF6E40",
            }
          }
          type="button"
          value="#FF6E40"
        />
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#448AFF",
            }
          }
          type="button"
          value="#448AFF"
        />
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#64FFDA",
            }
          }
          type="button"
          value="#64FFDA"
        />
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#B2FF59",
            }
          }
          type="button"
          value="#B2FF59"
        />
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#CDDC39",
            }
          }
          type="button"
          value="#CDDC39"
        />
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#795548",
            }
          }
          type="button"
          value="#795548"
        />
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#546E7A",
            }
          }
          type="button"
          value="#546E7A"
        />
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#0097A7",
            }
          }
          type="button"
          value="#0097A7"
        />
      </li>
      <li>
        <button
          className="undefined"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#EEEEEE",
            }
          }
          type="button"
          value="#EEEEEE"
        />
      </li>
    </ul>
    <div>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        <i
          className="material-icons"
        >
          clear
        </i>
      </button>
      <button
        className=""
        type="submit"
      >
        <i
          className="material-icons"
        >
          add
        </i>
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        <i
          className="material-icons"
        >
          color_lens
        </i>
      </button>
    </div>
  </form>
</div>
`;

exports[`EventForm component should render without problem 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div>
      <div>
        <input
          name="event"
          onChange={[Function]}
          required={true}
          style={
            Object {
              "borderBottom": "2px solid undefined",
            }
          }
          type="text"
          value=""
        />
        <label
          htmlFor="nodejs"
        >
          Rappel
        </label>
      </div>
    </div>
    <ul
      className=""
    >
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#FF5252",
            }
          }
          type="button"
          value="#FF5252"
        />
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#FF4081",
            }
          }
          type="button"
          value="#FF4081"
        />
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#E040FB",
            }
          }
          type="button"
          value="#E040FB"
        />
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#FF6E40",
            }
          }
          type="button"
          value="#FF6E40"
        />
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#448AFF",
            }
          }
          type="button"
          value="#448AFF"
        />
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#64FFDA",
            }
          }
          type="button"
          value="#64FFDA"
        />
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#B2FF59",
            }
          }
          type="button"
          value="#B2FF59"
        />
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#CDDC39",
            }
          }
          type="button"
          value="#CDDC39"
        />
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#795548",
            }
          }
          type="button"
          value="#795548"
        />
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#546E7A",
            }
          }
          type="button"
          value="#546E7A"
        />
      </li>
      <li>
        <button
          className=""
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#0097A7",
            }
          }
          type="button"
          value="#0097A7"
        />
      </li>
      <li>
        <button
          className="undefined"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#EEEEEE",
            }
          }
          type="button"
          value="#EEEEEE"
        />
      </li>
    </ul>
    <div
      className="undefined"
    >
      <i
        className="material-icons"
      >
        error
      </i>
      name
       
      is required
    </div>
    <div>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        <i
          className="material-icons"
        >
          clear
        </i>
      </button>
      <button
        className=""
        type="submit"
      >
        <i
          className="material-icons"
        >
          add
        </i>
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        <i
          className="material-icons"
        >
          color_lens
        </i>
      </button>
    </div>
  </form>
</div>
`;
